packages/radix/src/components/Checkbox.tsx

```diff
@@ -18,18 +18,18 @@ export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>((props
             width: theme.sizes[4],
             height: theme.sizes[4],
             borderRadius: theme.radii[1],
-            border: '1px solid',
-            borderColor: theme.colors.gray400,
+            border: 0,
+            boxShadow: `inset 0 0 0 1px ${theme.colors.gray400}`,
           },
           hover: {
-            borderColor: theme.colors.gray500,
+            boxShadow: `inset 0 0 0 1px ${theme.colors.gray500}`,
           },
           focus: {
-            borderColor: theme.colors.gray600,
+            boxShadow: `inset 0 0 0 1px ${theme.colors.blue500}, 0 0 0 1px ${theme.colors.blue500}`,
           },
           checked: {
             backgroundColor: theme.colors.blue600,
-            borderColor: theme.colors.blue600,
+            boxShadow: 'none',
```

Comment:
No focus state at all for checked?

Reply:
No, not at all. This is to have proper styles in normal state at least, without extra border. This didn’t have focus state before as well.

I think we either need to look at precedence of focus and checked states, or have an ability to style checked focus state.

On a related note, I’ve seen separate styles for checked hover and active states many times and I think we should support them as well.

Reply:
👍